components/claim/components/email_to_ldap.tsx

```diff
@@ -0,0 +1,233 @@
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import classNames from 'classnames';
+
+import {AuthChangeResponse} from '@mattermost/types/users';
+
+import {emailToLdap} from 'actions/admin_actions.jsx';
+
+import * as Utils from 'utils/utils';
+import {t} from 'utils/i18n.jsx';
+
+import LoginMfa from 'components/login/login_mfa';
+import LocalizedInput from 'components/localized_input/localized_input';
+
+type Props = {
+    email: string | null;
+    siteName?: string;
+    ldapLoginFieldName?: string;
+}
+
+const EmailToLDAP = ({email, siteName, ldapLoginFieldName}: Props) => {
+    const emailPasswordInput = useRef<HTMLInputElement>(null);
+    const ldapIdInput = useRef<HTMLInputElement>(null);
+    const ldapPasswordInput = useRef<HTMLInputElement>(null);
+
+    const [password, setPassword] = React.useState('');
+    const [ldapId, setLdapId] = React.useState<string | null>(null);
+    const [ldapPassword, setLdapPassword] = React.useState<string | null>(null);
+    const [passwordError, setPasswordError] = React.useState('');
+    const [ldapError, setLdapError] = React.useState('');
+    const [ldapPasswordError, setLdapPasswordError] = React.useState('');
+    const [serverError, setServerError] = React.useState('');
+    const [showMfa, setShowMfa] = React.useState(false);
+
+    const preSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+
+        const password = emailPasswordInput.current && emailPasswordInput.current.value;
+        if (!password) {
+            setPasswordError(Utils.localizeMessage('claim.email_to_ldap.pwdError', 'Please enter your password.'));
+            return;
+        }
+
+        const ldapId = ldapIdInput.current && ldapIdInput.current.value.trim();
+        if (!ldapId) {
+            setLdapError(Utils.localizeMessage('claim.email_to_ldap.ldapIdError', 'Please enter your AD/LDAP ID.'));
+            return;
+        }
+
+        const ldapPassword = ldapPasswordInput.current && ldapPasswordInput.current.value;
+        if (!ldapPassword) {
+            setLdapPasswordError(Utils.localizeMessage('claim.email_to_ldap.ldapPasswordError', 'Please enter your AD/LDAP password.'));
+            return;
+        }
+
+        setPassword(password);
+        setLdapId(ldapId);
+        setLdapPassword(ldapPassword);
+
+        if (email) {
+            submit(email, password, '', ldapId, ldapPassword);
+        }
+    };
+
+    const submit = (loginIdParam: string, passwordParam: string, token: string, ldapIdParam?: string, ldapPasswordParam?: string) => {
+        emailToLdap(
+            loginIdParam,
+            passwordParam,
+            token,
+            ldapIdParam || ldapId,
+            ldapPasswordParam || ldapPassword,
+            (data: AuthChangeResponse) => {
+                if (data.follow_link) {
+                    window.location.href = data.follow_link;
```

Comment:
In theory we should be able to use `useHistory`. Since I dont know how to test all these Claim components, I am not 100% sure. If you can guide me on testing, I can test before and after doing the change.

Reply:
is there no related test file?

Reply:
There is only one test file which is for 'ldap_to_email'. For the rest of components, there is no test

Reply:
would you mind to add tests for that as well? In a separate PR if course

Reply:
Yes, @michelengelen do you mind opening an issue and assign it to me? Just to have things under my radar and not forget them